ant/src/org/jetbrains/kotlin/ant/KotlinCompilerAdapter.java

```diff
@@ -87,15 +87,24 @@ public boolean execute() throws BuildException {
 
         kotlinc.getAdditionalArguments().addAll(additionalArguments);
 
-        kotlinc.execute();
-        if (!Integer.valueOf(0).equals(kotlinc.getExitCode())) {
-            // Don't run javac if failOnError = false and there were errors on Kotlin sources
-            return false;
+        // Javac13#execute passes everything in compileList to javac, which doesn't recognize .kt files
+        File[] compileListForJavac = filterOutKotlinSources(compileList);
+
+        boolean hasKotlinFilesInSources = compileListForJavac.length < compileList.length;
+
+        if (hasKotlinFilesInSources) {
+            kotlinc.execute();
+            if (!Integer.valueOf(0).equals(kotlinc.getExitCode())) {
+                // Don't run javac if failOnError = false and there were errors on Kotlin sources
+                return false;
+            }
+        }
+        else {
+            kotlinc.fillArguments();
```

Comment:
kotlinc.execute() invokes fillArgument(), if execute() is not invoked, then kotlinc.getArgs() will be empty. But ` addRuntimeToJavacClasspath(kotlinc)` uses kotlinc.getArgs().